tools/mercury/README.md

```diff
@@ -0,0 +1,508 @@
+
+You will also need to change that 'versions' file to point to one or more versions that you have installed on your system, and then change the 'test' file to only use your versions.  (If you only define one version, then it will be the default and will get used whenever you create a new router if you just don't use the 'version' arg in the 'routers' command.
+
+
+
+
+<br/>
+
+
+###Debugging Startup
+
+When Mercury starts up a router, it saves all the information you need to reproduce the same startup by hand. The router config file, the environment variables that are set, and the command line that is used are all saved in MERCURY\_ROOT/mercury/sessions/session\_TIMESTAMP/config. Router config files have the same names as their routers.
+
+Here is an example:
+
+    /home/mick/mercury/mercury/sessions/session\_2019\_03\_05\_2115/config/
+        |-- A.conf
+        |-- B.conf
+        |-- command_line
+        |-- environment_variables
+
+
+If you have a router fail to start, or it starts up and is immediately defunct, use this information to reproduce the same startup by hand, and see what's happening.
+
+
+
+<br/>
+
+
+###Running a test 'By Hand'
+
+One nice way to use Mercury is to use it to run a test for you and then see how it did that, so you can run the same setup 'by hand'.  You can look in the session/config directory and see all the environment variables it set and the command lines it used for the routers and the clients. 
+
+The command lines for the routers will point to the config files that it created, and those config files have ports that were chosen because they were free at that moment. It is possible that they will *no longer* be free when you run the test 'by hand' if you have other stuff running on your system. But unlikely.
+
+
+
+<br/>
+
+
+###Versions
+
+A 'version' represents a version of the dispatch router + proton code. The idea is that you have as many dispatch+proton installations as you like on your system, you define a Mercury version for each one of them, and then when you create a new router in Mercury you can tell it which version you want that router to use. Mercury will use the executable that corresponds to that version, and make it point to the correct libraries.
+
+You can define a version in one of two different ways
+
+1. You can provide the root directories for the proton and dispatch installation, and let Mercury calculate from them all the paths it needs, or
+
+2. You can directly provide all the paths. This second option is meant for situations where your installation is different somehow from what Mercury expects.
+
+To define a version with roots, use the 'version\_roots' command something like this:  (Here we define two different versions.)
+
+
+    version_roots name latest dispatch /home/mick/latest/install/dispatch proton /home/mick/latest/install/proton
+    version_roots name later  dispatch /home/mick/later/install/dispatch  proton /home/mick/later/install/proton
```

Comment:
Remove dispatch and proton and just assume that the first is dispatch path and second is proton path